src/test/java/com/microsoft/sqlserver/jdbc/datatypes/BigIntegerTest.java

```diff
@@ -0,0 +1,243 @@
+
+    enum TestType {
+        SETOBJECT_WITHTYPE, // This is to test conversions in Table B-5
+        SETOBJECT_WITHOUTTYPE, // This is to test conversions in Table B-4
+        SETNULL // This is to test setNull method
+    };
+
+    final static String tableName = RandomUtil.getIdentifier("BigIntegerTestTable");
+
+    @Test
+    public void testJDBC41BigInteger() throws Exception {
+        try (Connection conn = DriverManager.getConnection(connectionString); Statement stmt = conn.createStatement()) {
+
+            // Create the test table
+            try {
+                stmt.executeUpdate("drop table " + AbstractSQLGenerator.escapeIdentifier(tableName));
+            } catch (Exception e) {}
+
+            String query = "create table " + AbstractSQLGenerator.escapeIdentifier(tableName)
+                    + " (col1 varchar(100), col2 bigint, col3 real, col4 float, "
+                    + "col5 numeric(38,0), col6 int, col7 smallint, col8 char(100), col9 varchar(max), "
+                    + "id int IDENTITY primary key)";
+            stmt.executeUpdate(query);
+
+            try (PreparedStatement pstmt = conn
+                    .prepareStatement("INSERT INTO " + AbstractSQLGenerator.escapeIdentifier(tableName)
+                            + " VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?) SELECT * FROM "
+                            + AbstractSQLGenerator.escapeIdentifier(tableName) + " where id = ?")) {
+
+                // test that the driver converts the BigInteger values greater than LONG.MAX_VALUE and lesser than
+                // LONG.MIN_VALUE correctly
+                // A random value that is bigger than LONG.MAX_VALUE
+                BigInteger bigIntPos = new BigInteger("922337203685477580776767676");
+                // A random value that is smaller than LONG.MIN_VALUE
+                BigInteger bigIntNeg = new BigInteger("-922337203685477580776767676");
+
+                // Test the setObject method for different types of BigInteger values. Since BigInteger is mapped to
+                // JDBC
+                // BIGINT, the max and min limits for
+                int row = 1;
+                testSetObject(AbstractSQLGenerator.escapeIdentifier(tableName), BigInteger.valueOf(Long.MAX_VALUE),
+                        row++, pstmt, TestType.SETOBJECT_WITHTYPE);
+
+                testSetObject(AbstractSQLGenerator.escapeIdentifier(tableName), BigInteger.valueOf(Long.MIN_VALUE),
+                        row++, pstmt, TestType.SETOBJECT_WITHTYPE);
+                testSetObject(AbstractSQLGenerator.escapeIdentifier(tableName), BigInteger.valueOf(10), row++, pstmt,
+                        TestType.SETOBJECT_WITHTYPE);
+                testSetObject(AbstractSQLGenerator.escapeIdentifier(tableName), BigInteger.valueOf(-10), row++, pstmt,
+                        TestType.SETOBJECT_WITHTYPE);
+                testSetObject(AbstractSQLGenerator.escapeIdentifier(tableName), BigInteger.ZERO, row++, pstmt,

```

Comment:
Instead of multiple calls to `AbstractSQLGenerator.escapeIdentifier(tableName)` maintain `escapedTableName` instead?